clang/test/Driver/linker-wrapper.c

```diff
@@ -181,5 +181,6 @@ __attribute__((visibility("protected"), used)) int x;
 // RUN:   --linker-path=/usr/bin/ld.lld -- -r --whole-archive %t.a --no-whole-archive \
 // RUN:   %t.o -o a.out 2>&1 | FileCheck %s --check-prefix=RELOCATABLE-LINK
 
```

Comment:
need to check device fatbin bundling is done and device wrapper fatbin variable using internal linkage and in postfixed section.

Reply:
Added a test for the HIP and CUDA cases as well as an additional check on passing `-r` to the wrapper image generation.